users/views.py

```diff
@@ -97,33 +99,18 @@ def index(request: HttpRequest) -> HttpResponse:
     return render(request, "home.html", context={"num_visits": num_visits})
 
 
-class SignUp(APIView):
-    serializer_class = CustomRegisterSerializer
-    renderer_classes = [renderers.TemplateHTMLRenderer]
+class SignUp(FormView):
+    form_class = CustomUserSignUpForm
     template_name = "signup.html"
 
-    @classmethod
-    def get(cls, request: HttpRequest) -> Response:
-        logger.info(f"User get to the SignUp view with id: {request.user.pk}")
-        serializer = CustomRegisterSerializer(context={"request": request})
-        return Response({"serializer": serializer})
+    def form_valid(self, form: CustomUserSignUpForm) -> Union[Response, HttpResponseRedirectBase]:
+        user = form.save()
+        logger.info(f"New user with id: {user.pk} has been created")
+        return redirect("success-signup")
 
-    @classmethod
-    def post(cls, request: HttpRequest) -> Union[Response, HttpResponseRedirectBase]:
-        logger.info(f"User with id: {request.user.pk} sent post to the SignUp view")
-        serializer = CustomRegisterSerializer(data=request.data)
-        if not serializer.is_valid():
-            logger.debug(f"Sent form is invalid due to those errors: {serializer.errors}")
-            return Response(
-                {
-                    "serializer": serializer,
-                    "errors": serializer.errors,
-                    "non_field_errors": serializer.errors.get("non_field_errors"),
-                }
-            )
-        else:
-            serializer.save(request)
-            return redirect("login")
+
+class SuccessView(TemplateView):
```

Comment:
Please make this view name more descriptive, like `UserSignUpSuccess` or something similar.

Reply:
Done